src/app/pages/virtualization/components/all-instances/instance-details/instance-general-info/instance-general-info.component.html

```diff
@@ -21,6 +21,17 @@ <h3 mat-card-title>
       <p>{{ 'Base Image' | translate }}: {{ instance.image.description || '-' }}</p>
       <p>{{ 'CPU' | translate }}: {{ instance.cpu || '-' }}</p>
       <p>{{ 'Memory' | translate }}: {{ formatter.memorySizeFormatting(instance.memory) || '-' }}</p>
+
+      @if (instance.environment && objectKeys(instance.environment).length) {
```

Comment:
hey Alex I just found that `keyvalue` pipe could be used natively

https://github.com/truenas/webui/blob/21722ce6654cb70801715c47f98d67cc81cd9bd9/src/app/pages/apps/components/installed-apps/app-info-card/app-info-card.component.html#L88